includes/core/server/request/class-paginated.php

```diff
@@ -0,0 +1,154 @@
+<?php
+/**
+ * Class file for WCPay\Core\Server\Request\Paginated.
+ *
+ * @package WooCommerce Payments
+ */
+
+namespace WCPay\Core\Server\Request;
+
+use WCPay\Core\Exceptions\Server\Request\Invalid_Request_Parameter_Exception;
+use WCPay\Core\Server\Request;
+
+/**
+ * Request class for paginated requests.
+ */
+class Paginated extends Request {
+
+
+	const DEFAULTS = [
+		'page'      => 0,
+		'pagesize'  => 25,
+		'sort'      => 'created',
+		'direction' => 'desc',
+		'limit'     => 100,
+	];
+
+	/**
+	 * Request uri
+	 *
+	 * @var mixed (int|string)
+	 */
+	private $uri = null;
+
+	/**
+	 * Returns the request's API.
+	 *
+	 * @return string
+	 * @throws Invalid_Request_Parameter_Exception
+	 */
+	public function get_api(): string {
+		return $this->uri; // Will be changed on every request by overriding this method or use set_uri method.
```

Comment:
I wanted to avoid creating class, import it, create test for it and use it only in one place. It felt to me like we are creating a bunch of classes where only one line of code will be changed. But, I see your point here. Adding extra class is fine approach too, but considering that we have around 100+ API calls, we should be careful in creating a bunch of classes since it will be difficult to navigate through the code.

Reply:
> since it will be difficult to navigate through the code

IDEs for the win :)

If it's just the API method that's changing, we could have a single test for each of those sub-requests within a single test class, and just add one line there to prevent coverage decreases.

Reply:
This is refactored and all classes or requests have their own class with dedicated URI and other things liks filters and setters.